neurom/view/view.py

```diff
@@ -96,6 +84,27 @@ def get_default(variable, **kwargs):
     return kwargs.get(variable, default[variable])
 
 
+def _plane2col(plane):
+    '''take a string like 'xy', and return the indices from COLS.*'''
+    planes = ('xy', 'yx', 'xz', 'zx', 'yz', 'zy')
+    assert plane in planes, 'No such plane found! Please select one of: ' + str(planes)
+    return (getattr(COLS, plane[0].capitalize()),
+            getattr(COLS, plane[1].capitalize()), )
+
+
+def _get_linewidth(tr, parameters):
+    '''calculate the desired linewidth based on tree contents, and parameters'''
+    linewidth = get_default('linewidth', parameters)
+    # Definition of the linewidth according to diameter, if diameter is True.
+    if get_default('diameter', parameters):
+        # TODO: This was originally a numpy array. Did it have to be one?
+        scale = get_default('diameter_scale', parameters)
+        linewidth = [2 * segment_radius(s) * scale for s in iter_segments(tr)]
+        if not linewidth:
+            linewidth = get_default('linewidth', parameters)
+    return linewidth
+
+
 def tree(tr, plane='xy', new_fig=True, subplot=False, **kwargs):
```

Comment:
I would rather list the optional parameters with their defaults explicitly in function signatures (with defaults as constants to avoid code duplication).